File: app/src/main/java/com/mammoth/soft/huarongdao/utils/GameHRDDao.java
package com.mammoth.soft.huarongdao.utils;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

/**
 * Created by wuhaoyong on 02-03-2020.
 */
@Dao
public interface  GameHRDDao {
    @Query("SELECT * FROM hrd")
    List<GameHRD> getAll();

    /**
     * Update GameHRD
     */
    @Update
    void updateGame(GameHRD gameHRD);

    /**
     * Insert GameHRD
     */
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertGame(GameHRD gameHRD);
}

File: app/src/main/java/com/mammoth/soft/huarongdao/data/GameHRDDao.kt
package com.mammoth.soft.huarongdao.data

import androidx.room.*
import com.mammoth.soft.huarongdao.data.GameHRD

/**
 * Created by andy on 02-03-2020.
 */
@Dao
interface GameHRDDao {
    @get:Query("SELECT * FROM hrd")
    val all: List<GameHRD?>?

    /**
     * Update GameHRD
     */
    @Update
    fun updateGame(gameHRD: GameHRD?)

    /**
     * Insert GameHRD
     */
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertGame(gameHRD: GameHRD?)
}